docs/docs/porting-create-react-app-to-gatsby.md

```diff
@@ -0,0 +1,263 @@
+export default Foo
+```
+
+For more information about errors encountered during builds, see the doc on [debugging HTML builds](/docs/debugging-html-builds/)
+
+### Routing
+
+There are two possibilites of routes that you can setup: static and dynamic. A static route has content that doesn't change, whereas a dynamic route can [fetch data at runtime](/docs/client-data-fetching/) just like any other React app.
+
+Gatsby automatically turns React components in the pages folder into static routes.
+
+> **Note**: An advantage to having pages in separate files like this is a defined way of [automatically code splitting](/docs/how-code-splitting-works/), whereas Create React App requires you to use the `import()` syntax to assign what elements should be loaded dynamically
+
+For dynamic routes, you should implement routing with [@reach/router](https://reach.tech/router), which is already included with Gatsby. Dynamic routes can be implemented the same way you would implement a router in Create React App (or any other React application). However, because these routes won't be represented as HTML files in the final build, if you want users to be able to visit the routes directly (like entering the URL in the search bar), you'll need to generate pages in the `gatsby-node.js` file which is demonstrated in the [Building Apps with Gatsby](/docs/building-apps-with-gatsby/) guide.
+
+```jsx
+import React from "react"
+import { Router } from "@reach/router"
+
+const App = () => (
+  <Router>
+    <Route path="/user/" component={Users} />
+    <Route path="/user/:id" component={UserDetails} />
+  </Router>
+)
+
+export default App
+```
+
+Gatsby provides a `<Link />` component and a `navigate` function to help you direct users through pages on your site. You can read about how to use each in the [`gatsby-link` doc](/docs/gatsby-link/).
+
+### Handling state
+
+Because Gatsby rehydrates into a regular React app, state can be handled inside of components in the same way it would in Create React App. If you use a another library for state management and want to wrap your app in some sort of global state the section on [context providers](#context-providers) will be helpful.
+
+### Environment Variables
+
+Create React App requires you to create environment variables prefixed with `REACT_APP_`. Gatsby instead uses the `GATSBY_` prefix to [make environment variables accessible](/docs/environment-variables) in the browser context.
+
+```title=".env"
+# in Create React App
+REACT_APP_API_URL=http://someapi.com
+
+# in Gatsby
+GATSBY_API_URL=http://someapi.com
+```
+
+### Advanced customizations
+
+Part of Gatsby's philosophy around tooling is [progressively disclosing complexity](/docs/gatsby-core-philosophy/#progressively-disclose-complexity), this simplifies the experience for a wider audience while still allowing the option to configure more advanced features for those that feel inclined. You won't have to "eject" your Gatsby app to edit more complex configurations.
```

Comment:
```suggestion
Part of Gatsby's philosophy around tooling is [progressively disclosing complexity](/docs/gatsby-core-philosophy/#progressively-disclose-complexity); this simplifies the experience for a wider audience while still allowing the option to configure more advanced features for those that feel inclined. You won't have to "eject" your Gatsby app to edit more complex configurations.
```

Reply:
ah yes, the ol' joining of two independent clauses like I learned and then subsequently forgot about in English class back in the day